File: docs/guides/java/src/test/java/com/algolia/abTestImplementationChecklist.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;

public class abTestImplementationChecklist {

  private static String getUserToken() {
    // Implement your logic here
    return "";
  }

  public static void main(String[] args) throws Exception {
    SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY");

    // Set the searchParams and get the current user token
    SearchParamsObject searchParams = new SearchParamsObject().setQuery("User search query").setEnableABTest(true);

    String userToken = getUserToken();

    // Is the user token anonymous?
    if (userToken == null || userToken.isEmpty() || userToken.equals("YOUR_ANONYMOUS_USER_TOKEN")) {
      // Disable A/B testing for this request
      searchParams.setEnableABTest(false);
    } else {
      // Set the user token to the current user token
      searchParams.setUserToken(userToken);
    }

    try {
      // Perform the searchSingleIndex
      SearchResponse result = client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);
      // SearchSingleIndex results
      System.out.println(result);
    } catch (Exception err) {
      // SearchSingleIndex errors
      System.err.println(err);
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/abTestImplementationChecklist.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*

fun getUserToken(): String {
  // Implement your logic here
  return ""
}

suspend fun abTestImplementationChecklist() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  // Set the query and get the current user token
  val query = "User search query"
  val userToken = getUserToken()

  // Set the searchParams
  val searchParams: SearchParamsObject

  // Is the user token anonymous?
  if (userToken.isEmpty() || userToken == "YOUR_ANONYMOUS_USER_TOKEN") {
    // Disable A/B testing for this request
    searchParams = SearchParamsObject(query = query, enableABTest = false)
  } else {
    // Set the user token to the current user token
    searchParams = SearchParamsObject(query = query, userToken = userToken)
  }

  try {
    // Perform the searchSingleIndex
    val result =
      client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
    // SearchSingleIndex results
    println(result)
  } catch (err: Exception) {
    // SearchSingleIndex errors
    println(err)
  }
}
